lib/blacklight/solr/repository.rb

```diff
@@ -52,7 +52,7 @@ def send_and_receive(path, solr_params = {})
     protected
 
       def build_connection
-        RSolr.connect(connection_config)
+        RSolr.connect(connection_config.merge(adapter: connection_config[:http_adapter]))
```

Comment:
Nope; we're currently passing that value through to RSolr, even though it's a Blacklight-specific configuration. Until RSolr 2.x, the value was simply ignored, but in 2.x, we need to remove it.